tests/rest/client/v1/test_admin.py

```diff
@@ -366,47 +366,52 @@ def prepare(self, reactor, clock, hs):
         self.event_creation_handler = hs.get_event_creation_handler()
         hs.config.user_consent_version = "1"
 
-        self._consent_uri_builder = Mock()
-        self._consent_uri_builder.build_user_consent_uri.return_value = (
+        consent_uri_builder = Mock()
+        consent_uri_builder.build_user_consent_uri.return_value = (
             "http://example.com"
         )
+        self.event_creation_handler._consent_uri_builder = consent_uri_builder
 
         self.store = hs.get_datastore()
 
-    def test_shutdown_room_conset(self):
-        admin_user = self.register_user("admin", "pass", admin=True)
-        admin_user_tok = self.login("admin", "pass")
+        self.admin_user = self.register_user("admin", "pass", admin=True)
+        self.admin_user_tok = self.login("admin", "pass")
+
+        self.other_user = self.register_user("user", "pass")
+        self.other_user_token = self.login("user", "pass")
+
+        # Mark the admin user as having consented
+        self.get_success(
+            self.store.user_set_consent_version(self.admin_user, "1"),
+        )
 
-        other_user = self.register_user("user", "pass")
-        other_user_token = self.login("user", "pass")
+    def test_shutdown_room_conset(self):
```

Comment:
can you add a docstring that explains in slightly more words what you are testing for here?